keps/sig-autoscaling/0000-kep-template.md

```diff
@@ -0,0 +1,295 @@
+
+The canonical place for the latest set of instructions (and the likely source of this file) is [here](/keps/0000-kep-template.md).
+
+The `Metadata` section above is intended to support the creation of tooling around the KEP process.
+This will be a YAML section that is fenced as a code block.
+See the KEP process for details on each of these items.
+
+## Table of Contents
+
+TPB
+
+## Summary
+
+TPB
+
+## Motivation
+
+- delegate the responsibility of managing cloud providers out of the cluster autoscaler
+- in general, every tool has its own implementation of the cloud provider layer to communicate with cloud provider API (logic duplicated) => cluster-api project
+
+
+
+This section is for explicitly listing the motivation, goals and non-goals of this KEP.
+Describe why the change is important and the benefits to users.
+The motivation section can optionally provide links to [experience reports][] to demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+
+### Goals
+
+* cluster autoscaler is capable of scaling the cluster through machine API
+
+### Non-Goals
+
+* cluster autoscaler is able to autoprovision new node groups through machine API
+* cluster autoscaler is able to estimate node resource requirements through machine API
+* cluster autoscaler is able to use at least one pricing model on top of the machine API
+
+## Proposal
+
+Generalize the concept of node group autoscaling on the level of cloud provider
+by integrating with machine API layer of [sigs.k8s.io/cluster-api](https://github.com/kubernetes-sigs/cluster-api) project to
+build cloud provider-free implementation of cluster autoscaling mechanism
+([#1050](https://github.com/kubernetes/autoscaler/issues/1050)).
+
+Also suggest how to:
+* perform node autoprovision on the level of the machine API ([node autoprovisioning
+  upstream proposal](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/node_autoprovisioning.md)).
+* perform estimation of cluster autoscaling on the level of the machine API.
+* build various pricing models on top of the machine API.
+
+Cluster autoscaling builds on top of concept of node groups which effectively
+translates to cloud provider specific autoscaling groups, resp. autoscaling sets.
+In case of the machine API, the node group translates into machine set (living
+as native Kubernetes object).
+
+### Node group (scaling group)
+
+The autoscaler currently supports `AWS`, `Azure`, `GCE`, `GKE` providers.
+With `kubemark` as other cloud provider implementation that allows to create
+hollow kubernetes cluster with huge number of nodes over a small subset of real
+machines (for scaling testing purposes).
+Each of the currently supported cloud providers has its own concept of scaling groups:
+- AWS has autoscaling groups
+- Azure has autoscaling sets
+- GCE/GKE have autoscaling groups
```

Comment:
cluster api currently doesnt support GKE as i understand it . the integration of autoscaler with cluster api should be optional or opt in so that autoscaler continues to be usable in gke

Reply:
My understanding is that the implementation in Cluster Autoscaler would be done by adding a new 'Cluster API cloudprovider'. Existing cloudprovider integrations won't be impacted.